dsx-fw/DSX_Library/PWM/Src/PWM.c

```diff
@@ -1,47 +1,35 @@
 #include "PWM.h"
-#include "main.h"
-
 
 // Timer handles
 extern TIM_HandleTypeDef htim16;
 extern TIM_HandleTypeDef htim17;
 
 
-// Function for updating the PWM duty cycle
+// Function for updating the PWM duty cycle and starting the PWM
 void updateDutyCycle (TIM_HandleTypeDef htim, uint8_t dutyCycle)
```

Comment:
In your servo library, the writeServo function changes the ARR value to SERV_PERIOD. You need to change the ARR value  also in this updateDutyCycle function or else when you use writeServo and then updateDutyCycle , the output will not be right. I confirmed this on the scope.